### PR TITLE
[WIP] Refactoring

### PR DESCRIPTION
**Pull request by jtojnar**
I have  been trying to use `null` instead of `false` or `''` whenever possible. Additionally, I also changed majority of comparisons to strict version, filled in missing parameters to phpdoc and also added `array` type hints.

**Tests should be merged first.**

### CODE REVIEW

File: spouts/youtube/youtube.php [hunk 1]
```diff
@@ -91,11 +91,37 @@ public function getThumbnail() {
...
+     * taken from: http://zytzagoo.net/blog/2008/01/23/extracting-images-from-html-using-regular-expressions/
+     * Searches for the first occurence of an html <img> element in a string
+     * and extracts the src if it finds it. Returns null in case an <img>
+     * element is not found.
+     *
+     * @param string $html An HTML string
+     *
+     * @return ?string content of the src attribute of the first image
+     */
+    private function getImage($html) {
```
Comment (jtojnar) [inline spouts/youtube/youtube.php]: factor out to a helpers class

**Comment by jtojnar** [comment]
For some reason https://feeds.feedburner.com/sandraandwoo now uses feedburner favicon instead of the site one.

**Comment by niol** [comment]
Yes this happens since merging #920 .

**Comment by jtojnar** [comment]
You are right, I was probably using old master on my selfoss instance.

**Pull request merged**